src/screens/Home/index.js

```diff
@@ -1,24 +1,30 @@
-import React from 'react';
+import React, { useContext, useCallback } from 'react';
 import i18next from 'i18next';
 
+import { AuthContext } from '@components/ProtectedRoute';
 import logo from '@assets/logo.svg';
 import styles from './styles.module.scss';
 
 function Home() {
+  const context = useContext(AuthContext);
+
+  const handleLogout = useCallback(() => context.handleSetIsAuth(false), [
+    context,
```

Comment:
given that you're using it here, I would do it as I commented above. Here if the context reference changes this will trigger again, but you only want it to trigger if the function inside context changes. So It's better to separate it. ONe way is to so what I mentioned above and then the dependency will only be `handleSetIsAuth`